src/polyglot-notebooks-vscode-common/src/vscodeNotebookManagement.ts

```diff
@@ -31,13 +31,44 @@ export async function replaceNotebookCellMetadata(notebookUri: vscode.Uri, cellI
 }
 
 export async function replaceNotebookMetadata(notebookUri: vscode.Uri, documentMetadata: { [key: string]: any }): Promise<boolean> {
-    const notebookEdit = vscode.NotebookEdit.updateNotebookMetadata(documentMetadata);
-    const edit = new vscode.WorkspaceEdit();
-    edit.set(notebookUri, [notebookEdit]);
-    const succeeded = await vscode.workspace.applyEdit(edit);
-    return succeeded;
+    const notebook = vscode.workspace.notebookDocuments.find(d => d.uri === notebookUri);
+    if (notebook) {
+        const metadata = notebook.metadata;
+        const shouldUpdate = compare(metadata, documentMetadata);
+        if (shouldUpdate) {
+            const notebookEdit = vscode.NotebookEdit.updateNotebookMetadata(documentMetadata);
+            const edit = new vscode.WorkspaceEdit();
+            edit.set(notebookUri, [notebookEdit]);
+            const succeeded = await vscode.workspace.applyEdit(edit);
+            return succeeded;
+        }
+        return false;
+    }
+    return false;
 }
 
+function compare(object1: { [key: string]: any }, object2: { [key: string]: any }) {
+    const isObject = (object: any) => {
+        return object !== null && typeof object === 'object';
+    }
+    const objectKeys1 = Object.keys(object1);
+    const objectKeys2 = Object.keys(object2);
```

Comment:
very minor nitpick: object1Keys and object2Keys may read slightly better

Reply:
good point